code/game/objects/effects/landmarks_events.dm

```diff
@@ -0,0 +1,129 @@
+/*
+Landmark to be spawned by GMs and admins when running events.
+Do NOT directly spawn "" or buildmode spawn this.
+Use the "Manage Event Triggers" verb under "Eventkit" category.
+Admin verb is called by code\modules\admin\verbs\event_triggers.dm
+*/
+/obj/effect/landmark/event_trigger
+	name = "ONLY FOR EVENTS. DO NOT USE WHEN MAPPING"
+	desc = "Please notify staff if you see this!"
+	var/creator_ckey = ""
+	var/isTeamwork = FALSE //Whether to notify creator or whole team.
+	var/isRepeating = FALSE
+	var/coordinates = ""
+	var/cooldown = 0 //in seconds
+	var/last_trigger = 0
+	var/isLoud = FALSE
+	var/isNarrate = FALSE
+
+/obj/effect/landmark/event_trigger/New()
+	..()
+	coordinates = "(X:[loc.x];Y:[loc.y];Z:[loc.z])"
+	invisibility = 100
+
+/obj/effect/landmark/event_trigger/proc/set_vars(mob/M)
+	name = sanitize(tgui_input_text(M, "Input Name for the trigger", "Naming", "Event Trigger"))
+	creator_ckey = M.ckey
+	if(!event_triggers[creator_ckey])
+		event_triggers[creator_ckey] = list()
+	event_triggers[creator_ckey] |= list(src)
+	isTeamwork = (tgui_alert(M, "Notify rest of team?", "Teamwork", list("No", "Yes")) == "Yes" ? TRUE : FALSE)
+	if(!isTeamwork)
+		isLoud = (tgui_alert(M, "Should it make a bwoink when triggered for YOU?", "bwoink", list("No", "Yes")) == "Yes" ? TRUE : FALSE)
+	isRepeating = (tgui_alert(M, "Make it fire repeatedly?", "Repetition", list("No", "Yes")) == "Yes" ? TRUE : FALSE)
+	if(isRepeating)
+		cooldown = tgui_input_number(M, "Set cooldown in seconds. Minimum 5 seconds!", "Cooldown", 60, min_value = 5)
+		cooldown = cooldown SECONDS
+	else
+		delete_me = TRUE
+	log_admin("[M.ckey] has created a [isNarrate ? "Narrtion" : "Notification"] landmark trigger at [coordinates]")
+
+/obj/effect/landmark/event_trigger/Destroy()
+	if(event_triggers[creator_ckey])
+		event_triggers[creator_ckey] -= src
+	..()
+
+/obj/effect/landmark/event_trigger/Crossed(var/atom/movable/AM)
+	if(isobserver(AM) || !isliving(AM))
+		return FALSE
+	var/mob/living/L = AM
+	if(!L.ckey) return FALSE
+
+	if(world.time < (last_trigger + cooldown))
+		return FALSE
+	if(!isRepeating && last_trigger) //Used to avoid spam if qdel(src) fires too slowly
+		return FALSE
+	last_trigger = world.time
+
+	if(!creator_ckey)	//For some reason, the user didn't have a ckey. Let's clean up
+		qdel(src)
```

Comment:
Are you sure you want even narration landmarks to be deleted before actually narrating just because creator key is missing? Shouldn't happen normally, but still.

Reply:
Yes, because you cannot delete them using the management screen and it could be spammy for event chat if badly set.